### PR TITLE
feat: support for EC keys in JWK sets and EC JWTs when using JWKS

### LINKED ISSUE #719: [feature] JWKS should not be limited to RSA
**Describe the bug**. A clear and concise description of what the bug is.
The current code limits JWKS support to RSA only:
https://github.com/centrifugal/centrifugo/blob/master/internal/jwtverify/token_verifier_jwt.go#L205

ECC keys are becoming more and more prevalent, and with since the auth server providing the JWKS picks the algorithm, Centrifugo should support both RSA and ECC.

I will get a PR submitted for this.

### PR DESCRIPTION
**Pull request by shaunco**
test: new JWKS tests for RSA and EC
Fixes #719

### CODE REVIEW

File: internal/jwtverify/token_verifier_jwt.go [hunk 1]
```diff
@@ -211,17 +211,34 @@ func (j *jwksManager) verify(token *jwt.Token, tokenVars map[string]any) error {
...
+		verifier, err := jwt.NewVerifierES(jwt.Algorithm(spec.Algorithm), pubKey)
+		if err != nil {
+			return fmt.Errorf("%w: %s", errUnsupportedAlgorithm, spec.Algorithm)
+		}
+
+		return verifier.Verify(token)
 	}
 
-	return verifier.Verify(token)
+	return errUnsupportedAlgorithm
```
Comment (FZambia) [inline internal/jwtverify/token_verifier_jwt.go:241]: Minor thing: can we move this to switch default?

**Review by FZambia — APPROVED** [review]
Many thanks, great addition, and special 👍 for tests. 

Looking at your commit message starting with `fix: ` - it seems you still think it's a bug 😈 😀

**Comment by shaunco** [comment]
Fixed the PR title (😅) and moved that final return statement. I assume you will squash merge rather than wanting me to force push that last change, but let me know if I have that wrong.

**Comment by FZambia** [comment]
Thanks a lot, this is great!

**Comment by shaunco** [comment]
@FZambia - Thanks for getting this in! If you have a moment, a v5.0.5 release would be much appreciated!

**Comment by FZambia** [comment]
@shaunco , Centrifugo is usually released not often than once in 3-4 weeks, except bug fix releases. Most probably v5.0.5 will be released till the end of this week - requires some time to prepare.

**Comment by shaunco** [comment]
Thanks, I'll keep an eye out for it. We can't make use of it until the EC JWKS change is in a release.

**Pull request merged**